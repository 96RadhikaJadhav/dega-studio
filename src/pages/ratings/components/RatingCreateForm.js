import React from 'react';
import { Button, Form, Input, Space, InputNumber } from 'antd';
import { maker, checker } from '../../../utils/sluger';
import MediaSelector from '../../../components/MediaSelector';

const { TextArea } = Input;

const layout = {
  labelCol: {
    span: 10,
  },
  wrapperCol: {
    span: 14,
  },
};
const tailLayout = {
  wrapperCol: {
    offset: 10,
    span: 14,
  },
};

const RatingCreateForm = ({ onCreate, data = {} }) => {
  const [form] = Form.useForm();

  const onReset = () => {
    form.resetFields();
  };

  const onTitleChange = (string) => {
    form.setFieldsValue({
      slug: maker(string),
    });
  };

  return (
    <Form
      {...layout}
      form={form}
      initialValues={{ ...data }}
      name="creat-rating"
      onFinish={(values) => {
        onCreate(values);
        onReset();
      }}
    >
      <Form.Item
        name="name"
        label="Rating Name"
        rules={[
          {
            required: true,
            message: 'Please enter the name!',
          },
        ]}
      >
        <Input onChange={(e) => onTitleChange(e.target.value)} />
      </Form.Item>
      <Form.Item
        name="slug"
        label="Slug"
        rules={[
          {
            required: true,
            message: 'Please input the slug!',
          },
          {
            pattern: checker,
            message: 'Slug can not have whitespaces!',
          },
        ]}
      >
        <Input />
      </Form.Item>
      <Form.Item
        name="description"
        label="Description"
        rules={[
          {
            required: true,
            message: 'Please input the description!',
          },
        ]}
      >
        <TextArea />
      </Form.Item>
      <Form.Item name="numeric_value" label="numeric_value">
        <InputNumber min={1} max={5} defaultValue={3} />
      </Form.Item>
      <Form.Item label="Upload Media" name="medium_id">
        <MediaSelector />
      </Form.Item>
      <Form.Item {...tailLayout}>
        <Space>
          <Button type="primary" htmlType="submit">
            Submit
          </Button>
          <Button htmlType="button" onClick={onReset}>
            Reset
          </Button>
        </Space>
      </Form.Item>
    </Form>
  );
};

export default RatingCreateForm;
