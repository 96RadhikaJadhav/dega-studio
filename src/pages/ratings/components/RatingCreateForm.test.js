import React from 'react';
import renderer, { act } from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import '../../../matchMedia.mock';
import RatingCreateForm from './RatingCreateForm';

const data = {
  name: 'name',
  slug: 'slug',
  description: 'description',
  numeric_value: 3,
  medium_id: 1,
};

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

let onCreate, store;

describe('Ratings Create Form component', () => {
  beforeEach(() => {
    onCreate = jest.fn();
    onCreate.mockImplementationOnce((values) => new Promise((resolve, reject) => resolve(values)));
    store = mockStore({
      ratings: {
        req: [],
        details: {},
        loading: true,
      },
      media: {
        req: [],
        details: {},
        loading: true,
      },
    });
  });
  it('should render the component', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <RatingCreateForm />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should match component with empty data', () => {
    let component;
    act(() => {
      component = renderer.create(
        <Provider store={store}>
          <RatingCreateForm data={[]} />
        </Provider>,
      );
    });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should match component with data', () => {
    let component;
    act(() => {
      component = renderer.create(
        <Provider store={store}>
          <RatingCreateForm onCreate={onCreate} data={data} />
        </Provider>,
      );
    });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
