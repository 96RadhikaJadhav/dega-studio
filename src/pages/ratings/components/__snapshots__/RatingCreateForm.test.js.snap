// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ratings Create Form component should match component with data 1`] = `
<form
  className="ant-form ant-form-horizontal"
  id="creat-rating"
  onSubmit={[Function]}
>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-10 ant-form-item-label"
      style={Object {}}
    >
      <label
        className="ant-form-item-required"
        htmlFor="creat-rating_name"
        title="Rating Name"
      >
        Rating Name
      </label>
    </div>
    <div
      className="ant-col ant-col-14 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <input
            className="ant-input"
            id="creat-rating_name"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            type="text"
            value="name"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-10 ant-form-item-label"
      style={Object {}}
    >
      <label
        className="ant-form-item-required"
        htmlFor="creat-rating_slug"
        title="Slug"
      >
        Slug
      </label>
    </div>
    <div
      className="ant-col ant-col-14 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <input
            className="ant-input"
            id="creat-rating_slug"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            type="text"
            value="slug"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-10 ant-form-item-label"
      style={Object {}}
    >
      <label
        className=""
        htmlFor="creat-rating_description"
        title="Description"
      >
        Description
      </label>
    </div>
    <div
      className="ant-col ant-col-14 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <textarea
            className="ant-input"
            id="creat-rating_description"
            onChange={[Function]}
            onKeyDown={[Function]}
            style={Object {}}
            value="description"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-10 ant-form-item-label"
      style={Object {}}
    >
      <label
        className="ant-form-item-required"
        htmlFor="creat-rating_numeric_value"
        title="Numeric value"
      >
        Numeric value
      </label>
    </div>
    <div
      className="ant-col ant-col-14 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <div
            className="ant-input-number"
            onBlur={[Function]}
            onFocus={[Function]}
            style={Object {}}
          >
            <div
              className="ant-input-number-handler-wrap"
            >
              <span
                aria-label="Increase Value"
                className="ant-input-number-handler ant-input-number-handler-up"
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                role="button"
                unselectable="unselectable"
              >
                <span
                  aria-label="up"
                  className="anticon anticon-up ant-input-number-handler-up-inner"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="up"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M890.5 755.3L537.9 269.2c-12.8-17.6-39-17.6-51.7 0L133.5 755.3A8 8 0 00140 768h75c5.1 0 9.9-2.5 12.9-6.6L512 369.8l284.1 391.6c3 4.1 7.8 6.6 12.9 6.6h75c6.5 0 10.3-7.4 6.5-12.7z"
                    />
                  </svg>
                </span>
              </span>
              <span
                aria-label="Decrease Value"
                className="ant-input-number-handler ant-input-number-handler-down"
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                role="button"
                unselectable="unselectable"
              >
                <span
                  aria-label="down"
                  className="anticon anticon-down ant-input-number-handler-down-inner"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="down"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                    />
                  </svg>
                </span>
              </span>
            </div>
            <div
              className="ant-input-number-input-wrap"
            >
              <input
                aria-valuemax={5}
                aria-valuemin={1}
                aria-valuenow={3}
                autoComplete="off"
                className="ant-input-number-input"
                id="creat-rating_numeric_value"
                max={5}
                min={1}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseUp={[Function]}
                required={false}
                role="spinbutton"
                step={1}
                value="3"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-10 ant-form-item-label"
      style={Object {}}
    >
      <label
        className=""
        htmlFor="creat-rating_medium_id"
        title="Upload Media"
      >
        Upload Media
      </label>
    </div>
    <div
      className="ant-col ant-col-14 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <div
            className="ant-space ant-space-vertical"
          >
            <div
              className="ant-space-item"
              style={Object {}}
            >
              <button
                className="ant-btn"
                onClick={[Function]}
                type="button"
              >
                <span>
                  Select
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-14 ant-col-offset-10 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <div
            className="ant-space ant-space-horizontal ant-space-align-center"
          >
            <div
              className="ant-space-item"
              style={
                Object {
                  "marginRight": 8,
                }
              }
            >
              <button
                className="ant-btn ant-btn-primary"
                onClick={[Function]}
                type="submit"
              >
                <span>
                  Submit
                </span>
              </button>
            </div>
            <div
              className="ant-space-item"
              style={Object {}}
            >
              <button
                className="ant-btn"
                onClick={[Function]}
                type="button"
              >
                <span>
                  Reset
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</form>
`;

exports[`Ratings Create Form component should match component with empty data 1`] = `
<form
  className="ant-form ant-form-horizontal"
  id="creat-rating"
  onSubmit={[Function]}
>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-10 ant-form-item-label"
      style={Object {}}
    >
      <label
        className="ant-form-item-required"
        htmlFor="creat-rating_name"
        title="Rating Name"
      >
        Rating Name
      </label>
    </div>
    <div
      className="ant-col ant-col-14 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <input
            className="ant-input"
            id="creat-rating_name"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-10 ant-form-item-label"
      style={Object {}}
    >
      <label
        className="ant-form-item-required"
        htmlFor="creat-rating_slug"
        title="Slug"
      >
        Slug
      </label>
    </div>
    <div
      className="ant-col ant-col-14 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <input
            className="ant-input"
            id="creat-rating_slug"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-10 ant-form-item-label"
      style={Object {}}
    >
      <label
        className=""
        htmlFor="creat-rating_description"
        title="Description"
      >
        Description
      </label>
    </div>
    <div
      className="ant-col ant-col-14 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <textarea
            className="ant-input"
            id="creat-rating_description"
            onChange={[Function]}
            onKeyDown={[Function]}
            style={Object {}}
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-10 ant-form-item-label"
      style={Object {}}
    >
      <label
        className="ant-form-item-required"
        htmlFor="creat-rating_numeric_value"
        title="Numeric value"
      >
        Numeric value
      </label>
    </div>
    <div
      className="ant-col ant-col-14 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <div
            className="ant-input-number"
            onBlur={[Function]}
            onFocus={[Function]}
            style={Object {}}
          >
            <div
              className="ant-input-number-handler-wrap"
            >
              <span
                aria-label="Increase Value"
                className="ant-input-number-handler ant-input-number-handler-up"
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                role="button"
                unselectable="unselectable"
              >
                <span
                  aria-label="up"
                  className="anticon anticon-up ant-input-number-handler-up-inner"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="up"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M890.5 755.3L537.9 269.2c-12.8-17.6-39-17.6-51.7 0L133.5 755.3A8 8 0 00140 768h75c5.1 0 9.9-2.5 12.9-6.6L512 369.8l284.1 391.6c3 4.1 7.8 6.6 12.9 6.6h75c6.5 0 10.3-7.4 6.5-12.7z"
                    />
                  </svg>
                </span>
              </span>
              <span
                aria-label="Decrease Value"
                className="ant-input-number-handler ant-input-number-handler-down"
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                role="button"
                unselectable="unselectable"
              >
                <span
                  aria-label="down"
                  className="anticon anticon-down ant-input-number-handler-down-inner"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="down"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                    />
                  </svg>
                </span>
              </span>
            </div>
            <div
              className="ant-input-number-input-wrap"
            >
              <input
                aria-valuemax={5}
                aria-valuemin={1}
                autoComplete="off"
                className="ant-input-number-input"
                id="creat-rating_numeric_value"
                max={5}
                min={1}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseUp={[Function]}
                required={false}
                role="spinbutton"
                step={1}
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-10 ant-form-item-label"
      style={Object {}}
    >
      <label
        className=""
        htmlFor="creat-rating_medium_id"
        title="Upload Media"
      >
        Upload Media
      </label>
    </div>
    <div
      className="ant-col ant-col-14 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <div
            className="ant-space ant-space-vertical"
          >
            <div
              className="ant-space-item"
              style={Object {}}
            >
              <button
                className="ant-btn"
                onClick={[Function]}
                type="button"
              >
                <span>
                  Select
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-14 ant-col-offset-10 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <div
            className="ant-space ant-space-horizontal ant-space-align-center"
          >
            <div
              className="ant-space-item"
              style={
                Object {
                  "marginRight": 8,
                }
              }
            >
              <button
                className="ant-btn ant-btn-primary"
                onClick={[Function]}
                type="submit"
              >
                <span>
                  Submit
                </span>
              </button>
            </div>
            <div
              className="ant-space-item"
              style={Object {}}
            >
              <button
                className="ant-btn"
                onClick={[Function]}
                type="button"
              >
                <span>
                  Reset
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</form>
`;

exports[`Ratings Create Form component should render the component 1`] = `
<form
  className="ant-form ant-form-horizontal"
  id="creat-rating"
  onSubmit={[Function]}
>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-10 ant-form-item-label"
      style={Object {}}
    >
      <label
        className="ant-form-item-required"
        htmlFor="creat-rating_name"
        title="Rating Name"
      >
        Rating Name
      </label>
    </div>
    <div
      className="ant-col ant-col-14 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <input
            className="ant-input"
            id="creat-rating_name"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-10 ant-form-item-label"
      style={Object {}}
    >
      <label
        className="ant-form-item-required"
        htmlFor="creat-rating_slug"
        title="Slug"
      >
        Slug
      </label>
    </div>
    <div
      className="ant-col ant-col-14 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <input
            className="ant-input"
            id="creat-rating_slug"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-10 ant-form-item-label"
      style={Object {}}
    >
      <label
        className=""
        htmlFor="creat-rating_description"
        title="Description"
      >
        Description
      </label>
    </div>
    <div
      className="ant-col ant-col-14 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <textarea
            className="ant-input"
            id="creat-rating_description"
            onChange={[Function]}
            onKeyDown={[Function]}
            style={Object {}}
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-10 ant-form-item-label"
      style={Object {}}
    >
      <label
        className="ant-form-item-required"
        htmlFor="creat-rating_numeric_value"
        title="Numeric value"
      >
        Numeric value
      </label>
    </div>
    <div
      className="ant-col ant-col-14 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <div
            className="ant-input-number"
            onBlur={[Function]}
            onFocus={[Function]}
            style={Object {}}
          >
            <div
              className="ant-input-number-handler-wrap"
            >
              <span
                aria-label="Increase Value"
                className="ant-input-number-handler ant-input-number-handler-up"
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                role="button"
                unselectable="unselectable"
              >
                <span
                  aria-label="up"
                  className="anticon anticon-up ant-input-number-handler-up-inner"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="up"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M890.5 755.3L537.9 269.2c-12.8-17.6-39-17.6-51.7 0L133.5 755.3A8 8 0 00140 768h75c5.1 0 9.9-2.5 12.9-6.6L512 369.8l284.1 391.6c3 4.1 7.8 6.6 12.9 6.6h75c6.5 0 10.3-7.4 6.5-12.7z"
                    />
                  </svg>
                </span>
              </span>
              <span
                aria-label="Decrease Value"
                className="ant-input-number-handler ant-input-number-handler-down"
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                role="button"
                unselectable="unselectable"
              >
                <span
                  aria-label="down"
                  className="anticon anticon-down ant-input-number-handler-down-inner"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    className=""
                    data-icon="down"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                    />
                  </svg>
                </span>
              </span>
            </div>
            <div
              className="ant-input-number-input-wrap"
            >
              <input
                aria-valuemax={5}
                aria-valuemin={1}
                autoComplete="off"
                className="ant-input-number-input"
                id="creat-rating_numeric_value"
                max={5}
                min={1}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseUp={[Function]}
                required={false}
                role="spinbutton"
                step={1}
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-10 ant-form-item-label"
      style={Object {}}
    >
      <label
        className=""
        htmlFor="creat-rating_medium_id"
        title="Upload Media"
      >
        Upload Media
      </label>
    </div>
    <div
      className="ant-col ant-col-14 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <div
            className="ant-space ant-space-vertical"
          >
            <div
              className="ant-space-item"
              style={Object {}}
            >
              <button
                className="ant-btn"
                onClick={[Function]}
                type="button"
              >
                <span>
                  Select
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-14 ant-col-offset-10 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <div
            className="ant-space ant-space-horizontal ant-space-align-center"
          >
            <div
              className="ant-space-item"
              style={
                Object {
                  "marginRight": 8,
                }
              }
            >
              <button
                className="ant-btn ant-btn-primary"
                onClick={[Function]}
                type="submit"
              >
                <span>
                  Submit
                </span>
              </button>
            </div>
            <div
              className="ant-space-item"
              style={Object {}}
            >
              <button
                className="ant-btn"
                onClick={[Function]}
                type="button"
              >
                <span>
                  Reset
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</form>
`;
